tools/sense_studio/sense_studio.py

```diff
@@ -376,6 +370,49 @@ def show_video_list(split, label, path):
     return render_template('video_list.html', folders=folder_id, split=split, label=label, path=path)
 
 
+@app.route('/record-video/<string:project>/<string:split>/<string:label>')
+def record_video(project, split, label):
+    """
+    Display the video recording screen.
+    """
+    project = urllib.parse.unquote(project)
+    split = urllib.parse.unquote(split)
+    label = urllib.parse.unquote(label)
+    path = _lookup_project_path(project)
+    return render_template('video_recording.html', project=project, split=split, label=label, path=path)
+
+
+@app.route('/save-video/<string:project>/<string:split>/<string:label>', methods=['POST'])
+def save_video(project, split, label):
+    project = urllib.parse.unquote(project)
+    split = urllib.parse.unquote(split)
+    label = urllib.parse.unquote(label)
+    path = _lookup_project_path(project)
+
+    # Read given video to a file
+    input_stream = request.files['video']
+    output_path = os.path.join(path, f'videos_{split}', label)
+    temp_file_name = os.path.join(output_path, 'temp_video.webm')
+    with open(temp_file_name, 'wb') as temp_file:
+        temp_file.write(input_stream.read())
+
+    # Find a video name that is not used yet
+    existing_files = set(glob.glob(os.path.join(output_path, 'video_[0-9]*.mp4')))
+    video_idx = 0
+    output_file = os.path.join(output_path, f'video_{video_idx}.mp4')
+    while output_file in existing_files:
+        video_idx += 1
+        output_file = os.path.join(output_path, f'video_{video_idx}.mp4')
```

Comment:
I was positively surprised about this block of code when I deleted a video in the middle of the index range and found the newly recorded video took over the file name :)

Reply:
this might just create an issue :
if you delete a video without remove extracted features and frames, you will get an issue

Reply:
Good point @antoinemrcr. As part of the refactoring ticket I listed to add integrity tests when entering a dataset. Those test should probably also make sure there are no outdated features and frames.